### PR TITLE
A variety of cleanup/consistency releated to RAJA headers.

### PR DESCRIPTION
**Pull request by rhornung67**
A lot of minor things that will hopefully enable a smoother path for more involved reorg/cleanup in the future.

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
Why did you revert that commit? It should work if you mark the method `inline`?

**Comment by rhornung67** [comment]
It worked with all compilers except clang 5 and clang 6, including clang 4. Not sure why. Marking it inline doesn't seem like a portable solution.

**Comment by davidbeckingsale** [comment]
Why is inline non portable?

**Comment by rhornung67** [comment]
Because it's up to the compiler whether to inline or not. 'inline' is only a suggestion. I think the error is a linker issue, or possibly due to us setting flags to treat warnings as errors, but I didn't see that in the output.

**Comment by davidbeckingsale** [comment]
But the semantics of inline mean that multiple definitions of the function are okay as long as each translation unit gets an identical definition. (Regardless of whether the function is actually "inlined" by the compiler)

**Comment by rhornung67** [comment]
Yeah, I get that. Should I go back to what I did before and mark it inline?

File: include/RAJA/pattern/detail/forall.hpp [hunk 1]
```diff
@@ -1,3 +1,13 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief  Internal header for RAJA forall "BED" macros
```
[the remark this answers is not in the document]
Comment (rhornung67) [reply include/RAJA/pattern/detail/forall.hpp]: Added note that 'BED' stands for begin/end for loop bounds. I think that's what it means.

File: include/RAJA/internal/fault_tolerance.hpp [hunk 2]
```diff
@@ -38,9 +38,10 @@
 
 #include "RAJA/config.hpp"
 
-#ifdef RAJA_ENABLE_FT
+#if defined(RAJA_ENABLE_FT)
+
+#if definded(RAJA_REPORT_FT)
```
[the remark this answers is not in the document]
Comment (rhornung67) [reply include/RAJA/internal/fault_tolerance.hpp]: Thanks. Fixed.

File: include/RAJA/policy/tbb/forallN.hpp [hunk 3]
```diff
@@ -30,13 +30,16 @@
 
 #if defined(RAJA_ENABLE_TBB)
 
+#include <cstddef>
```
[the remark this answers is not in the document]
Comment (rhornung67) [reply include/RAJA/policy/tbb/forallN.hpp]: @trws I was following the Google C++ style guide recommendation
(https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes).
I can go either way on this, but at least all header inclusion follow the same order now. If you think putting system header file inclusions last, I will do that. I want to have a clear policy on this.

**Comment by trws** [comment]
You’re absolutely right, in fact it was my misremembering that same 
style guide that had me make the comment, go for it as it is in your 
updated changeset.  (apologies for the noise)

**Pull request merged**